Add support for Xapian Replication (resolves #1167)

sagepe opened the pull request:
Provides required support for Xapian Replication.

- Provides for setting `XAPIAN_MAX_CHANGESETS` in `conf/general` and uses export this into the environment within `search/index.pl` if present. This ensures it is set wherever the index script is called from.
- Supplies necessary scripts to integrate trigger replication from the client-side based on those used with Alaveteli.

Once this is merged, we activate by setting a values for the various new configuration variables in the config file. See also https://github.com/mysociety/sysadmin/issues/654.

dracos commented on search/replicate-xapian-database:
```diff
@@ -0,0 +1,67 @@
+REMOTEPATH="$5"
+IDENTITY_FILE="$6"
+
+if [ "x$8" != "x" ]; then
```
I think a couple of these var numbers are incorrect, due to the removal of the rails env one.

sagepe commented on search/index.pl:
```diff
@@ -31,6 +31,11 @@
+# If there's a value for XAPIAN_MAX_CHANGESETS in the config file,
+# make sure it's in our environment.
+my $changesets = mySociety::Config::get('XAPIAN_MAX_CHANGESETS');
+$changesets && $ENV{XAPIAN_MAX_CHANGESETS} = $changesets;
```
Apologies - I should have spotted this myself.

sagepe commented:
I've addresses the issues and made a stab at incorporating a call out to the replication script from `morningupdate` if the required config is present as suggested in https://github.com/mysociety/sysadmin/issues/654.

sagepe commented:
@struan or @mhl - would you mind taking a look at this please at some point?

It'a been refactored a bit since this PR was first opened. Are you happy with the approach, or would you rather take a difference tack? Happy to revisit if necessary. The issues mentioned by @dracos should be sorted, we've slightly changed the approach since then.
